### PR TITLE
L series support - RP changes

### PR DESCRIPTION
**Pull request by cadenmarchese**
## Which issue this PR addresses:

https://msazure.visualstudio.com/AzureRedHatOpenShift/_workitems/edit/10901562/

## What this PR does / why we need it:

Ls and Lsv2 family virtual machines offer local NVMe storage. More information about the new SKUs can be found [here](https://docs.microsoft.com/en-us/azure/virtual-machines/lsv2-series).

## Test plan for issue:

Confirm that the desired L series machines are available in the test region:
~~~
$ az vm list-skus --location westus --size Standard_L --all --output table
ResourceType     Locations    Name              Zones    Restrictions
---------------  -----------  ----------------  -------  --------------
virtualMachines  westus       Standard_L16s              None
virtualMachines  westus       Standard_L16s_v2           None
virtualMachines  westus       Standard_L32s              None
virtualMachines  westus       Standard_L32s_v2           None
virtualMachines  westus       Standard_L48s_v2           None
virtualMachines  westus       Standard_L4s               None
virtualMachines  westus       Standard_L64s_v2           None
virtualMachines  westus       Standard_L80s_v2           None
virtualMachines  westus       Standard_L8s               None
virtualMachines  westus       Standard_L8s_v2            None
~~~

### CLI Method

1) Using this branch, comment out line 44 in `pkg/env/dev.go`. This removes `FeatureRequireD2sV3Workers` from the list of feature flags and allows you to create development clusters with other worker VM sizes.

2) Follow the usual steps to [deploy a development RP](https://github.com/Azure/ARO-RP/blob/master/docs/deploy-development-rp.md), but don't use the hack script to create a cluster.

3) Follow steps in https://docs.microsoft.com/en-us/azure/openshift/tutorial-create-cluster to create a cluster, specifying `-worker-vm-size` in the `az aro create` step to specify an L series SKU:

~~~
az aro create   --resource-group $RESOURCEGROUP   --name $CLUSTER   --vnet aro-lseries   --master-subnet master-subnet   --worker-subnet worker-subnet   --worker-vm-size "Standard_L8s_v2"
~~~

### Hack scripts method

1) Using this branch, comment out line 44 in `pkg/env/dev.go` and modify line 411 of `pkg/util/cluster/cluster.go` to contain your desired instance size. For example:
~~~
oc.Properties.WorkerProfiles[0].VMSize = api.VMSizeStandardL4s
~~~

2) Use the [hack script to create a cluster.](https://github.com/cadenmarchese/ARO-RP/blob/master/docs/deploy-development-rp.md#run-the-rp-and-create-a-cluster)

### Day 2 Method

1)  Follow the usual steps to [deploy a development RP](https://github.com/Azure/ARO-RP/blob/master/docs/deploy-development-rp.md), using the hack script or the MSFT documentation to create a cluster. 

2) Edit the `vmSize` in a worker MachineSet to the desired L series SKU.

3) Delete the corresponding machine object (`oc delete machine`) . The updated MachineSet will provision a new one with desired instance type.
~~~
$ oc get machines
NAME                                  PHASE     TYPE              REGION   ZONE   AGE
cadenaro-m9ttf-master-0               Running   Standard_D8s_v3   eastus   1      40m
cadenaro-m9ttf-master-1               Running   Standard_D8s_v3   eastus   2      40m
cadenaro-m9ttf-master-2               Running   Standard_D8s_v3   eastus   3      40m
**cadenaro-m9ttf-worker-eastus1-86696   Running   Standard_L8s_v2   eastus   1      6m43s**
cadenaro-m9ttf-worker-eastus2-tb5hn   Running   Standard_D2s_v3   eastus   2      34m
cadenaro-m9ttf-worker-eastus3-szf9d   Running   Standard_D2s_v3   eastus   3      34m
~~~

### Is there any documentation that needs to be updated for this PR?

https://docs.microsoft.com/en-us/azure/openshift/support-policies-v4#supported-virtual-machine-sizes

### CODE REVIEW

File: pkg/api/admin/openshiftcluster.go
Comment (troy0820) [inline pkg/api/admin/openshiftcluster.go:164]: This size doesn't exists with the command:
```bash
az vm list-sizes -l $LOCATION -o table
```
The size of `Standard_L4s` does exist in westus2 but the docs do not include the 4 core vm in the v2 series listed here: 
https://docs.microsoft.com/en-us/azure/virtual-machines/lsv2-series.  _(Also it is recommended to use the newer vm types from the documentation)_

Are you including the previous generation version?

If you are including the **Standard_L4s** vm type this will need to change to `"Standard_L4s"`

File: pkg/api/openshiftcluster.go
Comment (troy0820) [inline pkg/api/openshiftcluster.go:280]: Same comment as above.

File: pkg/api/validate/dynamic/quota.go
Comment (troy0820) [inline pkg/api/validate/dynamic/quota.go:92]: Refer to comment [here](https://github.com/Azure/ARO-RP/pull/1751/files#r713123628)

File: pkg/api/admin/openshiftcluster.go
Comment (cadenmarchese) [reply pkg/api/admin/openshiftcluster.go:164]: Thanks for review. From what I can tell, the blocker is that the current `azure-sdk-for-go` doesn't yet contain all of the SKUs we intend to support (or any v2 SKUs):

https://github.com/Azure/azure-sdk-for-go/blob/main/services/compute/mgmt/2021-07-01/compute/enums.go#L1798-L1805

We have a request open with Microsoft to update it.

Comment (cadenmarchese) [reply pkg/api/admin/openshiftcluster.go:164]: Also, I see that the L4s are not available, and will also update accordingly.

Comment (troy0820) [reply pkg/api/admin/openshiftcluster.go:164]: Yeah there is no L4s_v2 and I'm not sure if you want to include the previous generation L4s vm type.  @jewzaam do we know if we want to do that?  
Should we put this on hold until we can get that answered?

**Comment by cadenmarchese** [comment]
Had a sync with @25region regarding this - it needs review, and if it looks good we can merge it in and then get the same SKU list into the new API skeleton.

**Review by s-amann — APPROVED** [review]

**Review by 25region — APPROVED** [review]
LGTM

**Review by bryanro92 — APPROVED** [review]
lgtm

**Comment by bryanro92** [comment]
after we cut the sprint 215 release we can merge this in shortly after so it can be rebased into the api skelly

**Review by m1kola — APPROVED** [review]
> ### Is there any documentation that needs to be updated for this PR?
>
> https://docs.microsoft.com/en-us/azure/openshift/support-policies-v4#supported-virtual-machine-sizes

Do we have a work item for this?

**Review by petrkotas — APPROVED** [review]
I am fine with this, I have one concern inderectly related to this PR. Will billing pick up the new machines automatically, or is there a need for an update?

Besides for this PR LGTM

**Comment by petrkotas** [comment]
@m1kola https://docs.microsoft.com/en-us/azure/openshift/support-policies-v4#day-2-worker-node I can already see them present. Which means, we should hurry and release it.

**Review by bennerv — APPROVED** [review]
lgtm

**Review by s-amann — APPROVED** [review]

**Pull request merged**
